hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -2235,6 +2235,17 @@ private void triggerFlushInPrimaryRegion(final HRegion region) {
       return;
     }
 
+    if (!region.getTableDescriptor().hasRegionMemStoreReplication()
```

Comment:
If our code logic reaches here, the value of  'WaitForPrimaryFlush' is always true, right? So we could only need to check `hasRegionMemStoreReplication`?

Reply:
Yes , you are right, and furthermore, we could merge this `if` to above `if `statement